### PR TITLE
NFDiv-2920 Add JS Order Granted Solicitor Cover Letter

### PR DESCRIPTION
**Pull request by BWeeks-JusticeGovUK**
https://tools.hmcts.net/jira/browse/NFDIV-2920

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java [hunk 1]
```diff
@@ -82,22 +109,76 @@ public void generateConditionalOrderPronouncedCoversheetOfflineRespondent(final
...
             applicant.getLanguagePreference(),
-            formatDocumentName(caseId, CONDITIONAL_ORDER_GRANTED_COVERSHEET_DOCUMENT_NAME, now(clock))
+            formatDocumentName(caseId, getCoverLetterDocumentName(caseData, applicant.isRepresented()), now(clock))
         );
     }
 
-    private Map<String, Object> templateVars(final CaseData caseData,
-                                             final Long caseId,
-                                             final Applicant applicant) {
+    private DocumentType getJSCoverLetterType(final boolean isRepresented, final boolean isDivorce, final boolean isOfflineRespondent) {
```
Comment (amoriartyMOJ) [inline src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java:131]: I don't believe we need to overwrite the coverletter type. It should remain as the original CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2. Same as we've done for JS flow on Notice of Proceedings. They remained as NOTICE_OF_PROCEEDINGS_APP1 / NOTICE_OF_PROCEEDINGS_APP2, even if solicitor / js / solicitor etc...

File: src/main/java/uk/gov/hmcts/divorce/document/model/DocumentType.java [hunk 2]
```diff
@@ -209,6 +209,30 @@ public enum DocumentType implements HasLabel {
     @JsonProperty("coGrantedCoversheetApp2")
     CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2("Respondent/Applicant 2 Conditional order granted cover letter"),
 
+    @JsonProperty("judicialSeparationGrantedCoversheet")
```
Comment (amoriartyMOJ) [inline src/main/java/uk/gov/hmcts/divorce/document/model/DocumentType.java:212]: As mentioned in other part of this review, we don't need to overwrite the type (as NoC for examples)

**Review by amoriartyMOJ — COMMENTED** [review]
Few comments around the same functionality changes. Any questions let me know happy to discuss!

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java (hunk 1 above)
Comment (BWeeks-JusticeGovUK) [reply src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java:131]: Yes, but then the document shows as the incorrect type in XUI.  It's not a conditional order document at that point, it's a judicial/separation document.  When caseworkers are going in and out of multiple cases a day, I believe misleading document types (and therefore descriptors) are going to make their jobs more difficult.

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java [hunk 3]
```diff
@@ -140,4 +191,36 @@ private Map<String, Object> templateVarsForOfflineRespondent(final CaseData case
...
+                                                           final Applicant applicant) {
+
+        final Map<String, Object> templateContent = docmosisCommonContent.getBasicDocmosisTemplateContent(
+            applicant.getLanguagePreference());
+
+        templateContent.put(SOLICITOR_NAME, applicant.getSolicitor().getName());
+        templateContent.put(SOLICITOR_FIRM, applicant.getSolicitor().getFirmName());
+        templateContent.put(SOLICITOR_ADDRESS, applicant.getSolicitor().getAddress());
+        templateContent.put(DATE, LocalDate.now().format(DATE_TIME_FORMATTER));
+        templateContent.put(CASE_REFERENCE, caseId != null ? formatId(caseId) : null);
```
Comment (FaisalMoJ) [inline src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java:206]: Is this check required? I believe caseId will never be null?

Comment (BWeeks-JusticeGovUK) [reply src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java:206]: You're probably correct, but that's how it's done elsewhere (for some reason).

**Review by FaisalMoJ — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java (hunk 3 above)
Comment (amoriartyMOJ) [reply src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java:204]: Yeah i dont think this is needed. Maybe done elsewhere from older code but case can't exist without one at this point

**Pull request merged**